src/library/scala/collection/mutable/AnyRefMap.scala

```diff
@@ -426,12 +426,12 @@ class AnyRefMap[K <: AnyRef, V] private[collection] (defaultEntry: K => V, initi
     this
   }
 
-  // The `K with AnyRef` parameter type is necessary to distinguish these methods from the base methods they overload (not override)
-  def map[K2 <: AnyRef, V2](f: ((K with AnyRef, V)) => (K2, V2)): AnyRefMap[K2, V2] =
+  // The implicit dummy parameter is necessary to distinguish these methods from the base methods they overload (not override)
```

Comment:
Can you at least mention the typer issue number in the comment, so that we can remove the workaround when the bug is fixed? (if it is…)